packages/web3-common/src/common_utils.ts

```diff
@@ -0,0 +1,306 @@
+import { keccak256 } from 'ethereum-cryptography/keccak';
+import { validator, isHexStrict, utils as validatorUtils, isAddress } from 'web3-validator';
+import { InvalidAddressError, InvalidBytesError, InvalidNumberError } from './errors';
+import { Address, Bytes, HexString, Numbers } from './types';
+
+const SHA3_EMPTY_BYTES = '0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470';
+
+export const bytesToBuffer = (data: Bytes): Buffer | never => {
```

Comment:
We have `web3-utils` package particularly for this purpose. The `web3-utils` should be used inside the `web3-common` not the other-way around. 

Think of it it as: 

1. `web3-utils` should be independent library consisting of the high level functions for small utilities. 
2. If we consider `web3` as a framework the `web3-common` is the common and abstract architectural components of the framework. 
3. It does not make sense to split the utility functions and in multiple packages and say `toWei` will be in `web3-utils` but `toBigInt` will be in `web3-common`.

Reply:
> I personally don't think the direction of the changes in this PR is correct and splitting up the utility functions among `web3-utils` and `web3-common` would be a nice idea. Or making a `web3-common` a dependency of the `web3-utils` would a right architectural choice.
> 
> Let's hear form others. @ChainSafe/web3-js

Its done for solving issue of cyclic dependency between utils and common. As there was wrong design of having two web3 Error base classes in existing code. 

web3-common, name suggests that its common functionality used in all or most of other packages, so there should be zero or minimum dependencies in web3-common.

Reply:
I don't think there was any cyclic dependency between `web3-utils` and `web3-common` packages. There are not just `Error` but few other duplicate codes and types in other packages as well to avoid the unnecessary dependency and that's not a bad design choice. And to improve such cases we can always create more domain specific packages e.g. in case of this PR `web3-errors` package will the right direction and that was discussed way earlier.

Reply:
using same base error class from web3 common in whole 4x created cyclic deps, and fix is in this pr.